samtranslator/model/eventsources/push.py

```
@@ -171,18 +204,39 @@ def to_cloudformation(self, **kwargs):
         events_rule = EventsRule(self.logical_id)
         events_rule.EventBusName = self.EventBusName
         events_rule.EventPattern = self.Pattern
-        events_rule.Targets = [self._construct_target(function)]
+        source_arn = events_rule.get_runtime_attr("arn")
+
+        dlq_queue_arn = None
+        if self.DeadLetterConfig is not None:
+            if "Arn" in self.DeadLetterConfig and "Type" in self.DeadLetterConfig:
+                raise InvalidEventException(
+                    self.logical_id, "You can either define Arn or Type property of DeadLetterConfig"
+                )
+
+            if "Arn" in self.DeadLetterConfig:
+                dlq_queue_arn = self.DeadLetterConfig["Arn"]
+            elif "Type" in self.DeadLetterConfig:
+                if self.DeadLetterConfig.get("Type") not in ["SQS"]:
+                    raise InvalidEventException(self.logical_id, "The only valid value for 'Type' is 'SQS'")
+                queue_logical_id = self.DeadLetterConfig.get("QueueLogicalId", None)
+                dlq_resources = EventBridgeRuleUtils.create_dead_letter_queue_with_policy(
+                    self.logical_id, source_arn, queue_logical_id
+                )
+                dlq_queue_arn = dlq_resources[0].get_runtime_attr("arn")
+                resources.extend(dlq_resources)
+            else:
+                raise TypeError("No Arn or Type provided")
```

Same as line 145, should raise `InvalidEventException` instead of `TypeError`